components/merino/src/merino.rs

```diff
@@ -0,0 +1,844 @@
+    pub schdeuled_corpus_item_id: Uuid,
+    pub url: Url,
+    pub title: String,
+    pub excerpt: String,
+    pub topic: Option<String>,
+    pub publisher: String,
+    #[serde(rename = "isTimeSensitive")]
+    pub is_time_sensitive: bool,
+    #[serde(rename = "imageUrl")]
+    pub image_url: Url,
+    #[serde(rename = "tileId")]
+    pub tile_id: i32,
+    #[serde(rename = "receivedRank")]
+    pub received_rank: i32,
+}
+
+// Ranked list of curated recommendations
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct CuratedRecommendationsBucket {
+    pub recommendations: Vec<ReccomendationDataItem>,
+    pub title: Option<String>,
+}
+
+// Fakespot product reccomendations
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct FakespotFeed {
+    pub products: Vec<FakespotProduct>,
+    #[serde(rename = "defaultCategoryName")]
+    pub default_category_name: String,
+    #[serde(rename = "headerCopy")]
+    pub header_copy: String,
+    #[serde(rename = "footerCopy")]
+    pub footer_copy: String,
+    pub cta: FakespotCta,
+}
+
+// Fakespot product details
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct FakespotProduct {
+    id: String,
+    title: String,
+    category: String,
+    #[serde(rename = "imageUrl")]
+    image_url: Url,
+    url: Url,
+}
+
+// Fakespot CTA
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct FakespotCta {
+    #[serde(rename = "ctaCopy")]
+    pub cta_copy: String,
+    pub url: Url,
+}
+
+// Ranked list of curated recommendations with responsive layout configs
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct FeedSection {
+    #[serde(rename = "receivedFeedRank")]
+    pub received_feed_rank: i32,
+    pub recommendations: Vec<ReccomendationDataItem>,
+    pub title: String,
+    pub subtitle: Option<String>,
+    pub layout: Layout,
+    #[serde(rename = "isFollowed")]
+    pub is_followed: bool,
+    #[serde(rename = "isBlocked")]
+    pub is_blocked: bool,
+}
+
+// Representation of a responsive layout configuration with multiple column layouts
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct Layout {
+    pub name: String,
+    #[serde(rename = "responsiveLayouts")]
+    pub responsive_layouts: Vec<ResponsiveLayout>,
+}
+
+// Layout configurations within a column
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct ResponsiveLayout {
+    #[serde(rename = "columnCount")]
+    pub column_count: i32,
+    pub tiles: Vec<Tile>,
+}
+// Properties for a single tile in a responsive layout
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct Tile {
+    pub size: String,
+    pub position: i32,
+    #[serde(rename = "hasAd")]
+    pub has_ad: bool,
+    #[serde(rename = "hasExcerpt")]
+    pub has_excerpt: bool,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde_json;
+
+    #[test]
+    fn test_curated_reccomendations_request_deserialization() {
```

Comment:
I hesitate to say this, but do we need these unit tests?  It feels like it's testing the serde code more than anything.    I could be missing something though, I'll let you make the call on if you think they're needed or not.